File: app/src/main/java/com/kip/receiptscanner/LauncherManager.java
package com.kip.receiptscanner;

import android.content.Context;
import android.content.SharedPreferences;

public class LauncherManager {
    SharedPreferences sharedPreferences;
    SharedPreferences.Editor editor;

    private static String PREF_NAME = "LaunchManager";
    private static String IS_FIRST_TIME = "isFirst";

    public LauncherManager(Context context){
        sharedPreferences = context.getSharedPreferences(PREF_NAME,0);
        editor = sharedPreferences.edit();
    }

    public void setFirstLaunch(boolean isFirst){
        editor.putBoolean(IS_FIRST_TIME,isFirst);
        editor.commit();
    }

    public boolean isFirstTime(){
        return sharedPreferences.getBoolean(IS_FIRST_TIME,true);
    }
}

File: app/src/main/java/com/kip/receiptscanner/LauncherManager.kt
package com.kip.receiptscanner

import android.content.Context
import android.content.SharedPreferences

class LauncherManager(context: Context) {
    var sharedPreferences: SharedPreferences
    var editor: SharedPreferences.Editor
    fun setFirstLaunch(isFirst: Boolean) {
        editor.putBoolean(IS_FIRST_TIME, isFirst)
        editor.commit()
    }

    val isFirstTime: Boolean
        get() = sharedPreferences.getBoolean(IS_FIRST_TIME, true)

    companion object {
        private const val PREF_NAME = "LaunchManager"
        private const val IS_FIRST_TIME = "isFirst"
    }

    init {
        sharedPreferences = context.getSharedPreferences(PREF_NAME, 0)
        editor = sharedPreferences.edit()
    }
}